pkgs/by-name/vs/vscode-extensions-update/vscode_extensions_update.py

```diff
@@ -0,0 +1,263 @@
+                )
+                try:
+                    self.execute_command(
+                        f"semver {self.target_vscode_version} -r '{engine_version_constraint}'"
+                    )
+                    print(f"Compatible version found: {candidate_version}")
+                    return candidate_version
+                except (ValueError, subprocess.CalledProcessError):
+                    print(
+                        f"Version {candidate_version} is not compatible with VSCode {self.target_vscode_version} (constraint: {engine_version_constraint})."
+                    )
+                continue
+            return candidate_version
+        print(
+            "Error: not found compatible version.",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+    def run_nix_update(self, new_version: str, system: str) -> None:
+        """
+        Builds and executes the nix-update command.
+        """
+        if not self.override_filename:
+            self.override_filename = self.execute_command(
+                f'EDITOR=echo nix edit --extra-experimental-features nix-command -f . "{self.attribute_path}"'
+            )
+            if (
+                "pkgs/applications/editors/vscode/extensions/vscode-utils.nix"
+                in self.override_filename
+            ):
+                self.override_filename = (
+                    "pkgs/applications/editors/vscode/extensions/default.nix"
+                )
+        update_command = f'nix-update "{self.attribute_path}" --version "{new_version}" --override-filename "{self.override_filename}" --system "{system}"'
+        print(f"Running update command:\n  {update_command}")
+        self.execute_command(update_command)
+
+    def run(self):
+        marketplace_data = self.get_marketplace_extension_data(
+            self.extension_marketplace_id
+        )
+        available_versions = marketplace_data.get("versions", [])
+        print(
+            f"Total versions found for {self.extension_marketplace_id}: {len(available_versions)}"
+        )
+        new_version = self.find_compatible_extension_version(
+            available_versions, self.get_target_platform(self.systems[0])
+        )
+        try:
+            self.execute_command(f"semver {self.current_version} -r '<{new_version}'")
+        except subprocess.CalledProcessError:
+            print("Already up to date or new version is older!")
+            sys.exit(0)
+        print(f"Selected extension version from Marketplace: {new_version}")
+        for i, system in enumerate(self.systems):
+            version = new_version if i == 0 else "skip"
+            self.run_nix_update(version, system)
+
+
+if __name__ == "__main__":
+    if not os.path.exists("nixos/release.nix"):
+        print("Please run this script in the nixpkgs root directory.", file=sys.stderr)
+        sys.exit(1)
```

Comment:
I would check this inside `VSCodeExtensionUpdater.__init__` and add a flag to skip the check